src/super_gradients/training/sg_model/sg_model.py

```diff
@@ -225,6 +225,7 @@ def connect_dataset_interface(self, dataset_interface: DatasetInterface, data_lo
         self.dataset_params = self.dataset_interface.get_dataset_params()
 
     # FIXME - we need to resolve flake8's 'function is too complex' for this function
+    @deprecated(version='2.2.0', reason="directly pass net (nn.Module) to train()")
```

Comment:
lets change the deprecation library - this one has really unclear api

Reply:
fixed